### PR TITLE
Add language icons to treeview

### LINKED ISSUE #65: [FEATURE] Icon depending on programming language of exercise and book icon for reading activity
Currently, there is no clear distinction between reading activities and exercises in the extension (only when hovering over the name). This proposal is similar to the UI on Dodona but in one column instead of two (with priority for the correct/wrong circle).

image image

This proposal uses the same icons as Dodona for programming exercises: https://dodona.ugent.be/nl/programming_languages/ (with a different icon for each programming language). These icons (programming language icon or book) would only be shown until the student did the activity (with correct or wrong as a result). The correct or wrong icons still have priority, because the programming language or book icon is less relevant if the student already made the exercise or did the reading activity. 

This is a mock-up of how it could look like:
image 

These are the [icons](https://iconify.design/) used on Dodona:
- bash.svg
- book-half.svg (icon for reading activities)
- file-document-edit-outline.svg (default if there are new programming languages added in the future)
- language-c.svg
- language-csharp.svg
- language-haskell.svg
- language-java.svg
- language-javascript.svg
- [language-python.svg](https://iconify.design/icon-sets/mdi/language-python.html)
- language-r.svg
- owl.svg
- language-kotlin.svg

### PR DESCRIPTION
**Pull request by BTWS2**
Before:
image
After:
image
For reference: https://dodona.ugent.be/nl/programming_languages/.
Icons: https://iconify.design/
Closes #65.

### CODE REVIEW

File: src/treeView/items/activityTreeItem.ts [hunk 1]
```diff
@@ -106,8 +114,16 @@ class ExerciseTreeItem extends AbstractActivityTreeItem {
...
+                `exercise-${status}.svg`);
+        } else {
+            this.iconPath = path.join(
+                __filename,
+                "..",
+                "..",
+                "..",
+                "..",
+                "assets",
+                `language-${String(exercise.programming_language?.name)}.svg`);
```
Comment (thepieterdc) [inline src/treeView/items/activityTreeItem.ts:126]: This needs a check in case the programming language is null.

**Review by thepieterdc — CHANGES_REQUESTED** [review]

File: src/treeView/items/activityTreeItem.ts [hunk 2]
```diff
@@ -106,8 +114,25 @@ class ExerciseTreeItem extends AbstractActivityTreeItem {
...
+                `language-text.svg`);
+        } else {
+            this.iconPath = path.join(
+                __filename,
+                "..",
+                "..",
+                "..",
+                "..",
+                "assets",
+                `language-${String(exercise.programming_language?.name)}.svg`);
```
Comment (stijndcl) [inline src/treeView/items/activityTreeItem.ts:135]: I believe this should work. `?.` returns undefined if the value to the left was null, and `first || second` should return `second` if the value to the left is falsy. Avoids about 10 lines of code duplication. Also, is the `String()`-conversion required here? `programming_language.name` is already a `string` as far as I can see.
```suggestion
        } else {
            this.iconPath = path.join(
                __filename,
                "..",
                "..",
                "..",
                "..",
                "assets",
                `language-${String(exercise.programming_language?.name || "text")}.svg`);
```

**Review by stijndcl — CHANGES_REQUESTED** [review]

File: src/treeView/items/activityTreeItem.ts (hunk 2 above)
Comment (BTWS2) [reply src/treeView/items/activityTreeItem.ts]: Thank you, I tried the shorter syntax and it works.

File: src/treeView/items/activityTreeItem.ts [hunk 3]
```diff
@@ -106,8 +114,16 @@ class ExerciseTreeItem extends AbstractActivityTreeItem {
...
+                `exercise-${status}.svg`);
+        } else {
+            this.iconPath = path.join(
+                __filename,
+                "..",
+                "..",
+                "..",
+                "..",
+                "assets",
+                `language-${exercise.programming_language?.name + ".svg" || "text.svg"}`);
```
Comment (thepieterdc) [inline src/treeView/items/activityTreeItem.ts:126]:
```suggestion
                `language-${exercise.programming_language?.name || "text"}.svg`);
```

**Review by thepieterdc — APPROVED** [review]

**Pull request merged**